### PR TITLE
Auto-fuzz: Extract base file templates and add project language and build type detection logic

### PR DESCRIPTION
**Pull request by arthurscchan**
This PR changes the logic for base project file generation and detection of project build type. Templates for those base project has been extracted to the template directory. The new logic will detect the language and project build type of the target projects. It will use the information to retrieve and generate correct base project files which also configure the JDK versions.

### CODE REVIEW

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]
Could you add a bit more description in the PR on what this changes?

Also, does `project_type` only reflect which build tool is used? If so, then I think we should adjust the name to something related to `build`.

File: tools/auto-fuzz/manager.py [hunk 1]
```diff
@@ -135,27 +135,37 @@ def project_name(self):
...
-    def write_basefiles(self):
+    def write_basefiles(self, project_build_type=None):
         with open(self.build_script, "w") as bfile:
-            bfile.write(base_files.gen_builder_1(self.language))
+            bfile.write(
+                base_files.gen_builder_1(self.language, project_build_type))
 
         with open(self.base_fuzzer, "w") as ffile:
-            ffile.write(base_files.gen_base_fuzzer(self.language))
+            ffile.write(
```
Comment (DavidKorczynski) [inline tools/auto-fuzz/manager.py:144]: Could we use a better name than `ffile`?

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]

File: tools/auto-fuzz/manager.py (hunk 1 above)
Comment (arthurscchan) [reply tools/auto-fuzz/manager.py]: Changed..

**Review by DavidKorczynski — APPROVED** [review]
LGTM, nice refactoring!

**Pull request merged**